### PR TITLE
docs: add natural language postgres guide

### PR DESCRIPTION
**Pull request by nicoalbanese**

### CODE REVIEW

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 1]
```diff
@@ -0,0 +1,683 @@
+---
+title: Natural Language Postgres
+description: Learn how to build a Next.js app that uses AI to interact with a PostgreSQL database using natural language.
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:3]: "Learn how to build a Next.js app that lets you talk to a PostgreSQL database in natural language" - potential to make it more direct (also below)

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+---
+
+# Natural Language Postgres Guide
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:6]: What would be the best name for SEO? What would users type? Can we include DB to make it broader?

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+
+In this guide, you will learn how to build an app that uses AI to interact with a PostgreSQL database using natural language.
+
+The application will use AI to:
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:10]: You can cut AI above if you mention it here imo

File: content/docs/02-guides/06-natural-language-postgres.mdx (hunk 2 above)
Comment (nicoalbanese) [reply content/docs/02-guides/06-natural-language-postgres.mdx:6]: Talk to DB?

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 4]
```diff
@@ -0,0 +1,683 @@
...
+```
+
+There are several important elements of this prompt:
+
+- Schema description helps the model understand exactly what data fields to work with
+- Includes rules for handling queries based on common SQL patterns - for example, always using ILIKE for case-insensitive string matching
+- Explains how to handle edge cases in the dataset, like dealing with the comma-separated investors field and ensuring whitespace is properly handled
+- Instead of having the model guess at industry categories, it provides the exact list that exists in the data, helping avoid mismatches
+- The prompt helps standardize data transformations - like knowing to interpret "10b" as "10.0" billion dollars, or that rates should be decimal values
+- Clear rules ensure the query output will be chart-friendly by always including at least two columns of data that can be plotted
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:211]: Love it

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+
+In this action, you'll use the `generateObject` function from the AI SDK which allows you to constrain the model's output to a pre-defined schema. This process, sometimes called structured output, ensures the model returns only the SQL query without any additional prefixes, explanations, or formatting that would require manual parsing.
+
+```ts filename="app/actions.ts"
+// other imports
+import { generateObject } from 'ai';
+import { openai } from '@ai-sdk/openai';
+import { z } from 'zod';
+
+// rest of the file
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:237]: comments here and above needed? or maybe have `// ... rest of file` (it was not immediately clear to me what the comments meant)

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+    setLoadingStep(2);
+
+    const companies = await runGeneratedSQLQuery(query);
+    const columns = companies.length > 0 ? Object.keys(companies[0]) : [];
+    setResults(companies);
+    setColumns(columns);
+
+    setLoading(false);
+
+    // TODO: Generate chart config & update chart config state with result
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:304]: on purpose? maybe `// SHOWN BELOW: ...`

**Review by lgrammel — APPROVED** [review]

File: content/docs/02-guides/06-natural-language-postgres.mdx [hunk 7]
```diff
@@ -0,0 +1,674 @@
...
+3. Generate a chart configuration for the results
+4. Allow toggling between table and chart views
+
+Head back to the browser and test the application with a few queries. You should see the chart visualization appear after the table results.
+
+
+### Next Steps
+You've built an AI-powered SQL analysis tool that can convert natural language to SQL queries, visualize query results, and explain SQL queries in plain English.
+
+Consider extending the application by using your own data sources or adding more advanced features like customizing the chart configuration schema to support more chart types and options and adding more complex SQL query generation capabilities.
```
Comment (lgrammel) [inline content/docs/02-guides/06-natural-language-postgres.mdx:674]: `You could, for example, extend the application...`
